runner/run_benchmark.py

```diff
@@ -25,61 +25,60 @@
-    elif name == "langchain_kg_a":
-        return LangchainAgent(model_name="gpt-3.5-turbo-instruct", mem_type=LangchainMemType.KG, **kwargs)
-    elif name == "langchain_ce_a":
-        return LangchainAgent(
-            model_name="gpt-3.5-turbo-instruct", mem_type=LangchainMemType.CONVERSATION_ENTITY, **kwargs
-        )
-    elif name == "ltm_agent_1":
-        return LTMAgentWrapper(model="gpt-4-0125-preview",
-                               variant=LTMAgentVariant.QG_JSON_USER_INFO, **kwargs)
-    elif name == "ltm_agent_2":
-        return LTMAgentWrapper(model="gpt-4-0125-preview",
-                               variant=LTMAgentVariant.SEMANTIC_ONLY, **kwargs)
-    elif name == "ltm_agent_3":
-        return LTMAgentWrapper(model="gpt-4-0125-preview",
-                               variant=LTMAgentVariant.TEXT_SCRATCHPAD, **kwargs)
-    elif name == "length_bias":
-        return LengthBiasAgent(model="gpt-4-0125-preview", **kwargs)
-    elif name.startswith("cost("):
+    if name.startswith("langchain_"):
+        suffix = "_".join(name.split("_")[1:])
+        mem_type = dict(
+            sb_a=LangchainMemType.SUMMARY_BUFFER,
+            kg_a=LangchainMemType.KG,
+            ce_a=LangchainMemType.CONVERSATION_ENTITY,
+        ).get(suffix, None)
+        if mem_type is None:
+            raise ValueError(f"Unrecognized LangChain memory type {repr(suffix)}.")
+        return LangchainAgent(model_name="gpt-3.5-turbo-instruct", mem_type=mem_type, **kwargs)
+    if name.startswith("ltm_agent_"):
+        agent_nr = "_".join(name.split("_")[2:])
+        variant = {
+            "1": LTMAgentVariant.QG_JSON_USER_INFO,
+            "2": LTMAgentVariant.SEMANTIC_ONLY,
+            "3": LTMAgentVariant.TEXT_SCRATCHPAD,
+        }.get(agent_nr, None)
+        if variant is None:
+            raise ValueError(f"Unrecognized LTM Agent variant {repr(agent_nr)}.")
+        return LTMAgentWrapper(model=GPT_4_TURBO_BEST, variant=variant, **kwargs)
+    if name == "length_bias":
+        return LengthBiasAgent(model=GPT_4_TURBO_BEST, **kwargs)
+    if name.startswith("cost("):
         in_cost, out_cost = [float(p.strip()) / 1_000 for p in name.removeprefix("cost(").removesuffix(")").split(",")]
         return CostEstimationChatSession(cost_in_token=in_cost, cost_out_token=out_cost, **kwargs)
-    elif name == "claude-2.1":
-        return ClaudeChatSession(**kwargs)
-    elif name == "claude-3-sonnet":
-        return ClaudeChatSession(**kwargs, model="claude-3-sonnet-20240229")
-    elif name == "claude-3-opus":
-        return ClaudeChatSession(**kwargs, model="claude-3-opus-20240229")
-    elif name == "human":
+    if name == "human":
         return HumanChatSession(**kwargs)
-    else:
-        raise ValueError(f"Unrecognized agent: {name}")
+
+    try:
+        model = get_model(name.removesuffix("ts-"))
+        if model.startswith("claude-"):
+            assert not name.startswith("ts-")
+            cls = ClaudeChatSession
```

Comment:
Why is this check for Claude in the `ts` section?. And also, why not just check for `startswith("ts")` and branch from there?

Reply:
The timestamped interface currently supports only OpenAI models.

Reply:
So this is a defence against someone doing `ts-claude`? I think that is unessecary since we know that it will fail immediately afterward. Would it not be better to just have a match for the name on `claude`, then do the whole `ts` parsing for the timestamped bit. It feels a little convoluted that  in order to select a claude model, you have to make sure that its not some other model first. 

Also just noticed, the function `removesuffix` should be `removeprefix`